images/velero-plugin-for-csi/main.tf

```diff
@@ -0,0 +1,62 @@
+terraform {
+  required_providers {
+    oci = { source = "chainguard-dev/oci" }
+  }
+}
+
+locals {
+  components = toset(["velero-plugin-for-csi"])
+}
+
+variable "target_repository" {
+  description = "The docker repo into which the image and attestations should be published."
+}
+
+module "config" {
+  for_each       = local.components
+  source         = "./config"
+  extra_packages = [each.key, "${each.key}-compat"]
+}
+
+module "velero-plugin-for-csi" {
+  for_each          = local.components
+  source            = "../../tflib/publisher"
+  name              = basename(path.module)
+  target_repository = var.target_repository
+  config            = module.config[each.key].config
+
+  build-dev    = true
+  main_package = each.key
+}
+
+data "oci_ref" "velero" {
+  ref = "cgr.dev/chainguard/velero:latest"
+}
+
+data "oci_ref" "velero-plugin-for-aws" {
+  ref = "cgr.dev/chainguard/velero-plugin-for-aws:latest"
+}
+
+module "test" {
+  source = "./tests"
+  digests = {
+    velero                = "cgr.dev/chainguard/velero:latest@${data.oci_ref.velero.digest}"
```

Comment:
Nit - but could we use "${data.oci_ref.velero.ref}@${data.oci_ref.velero.digest}" here to load in the registry URL instead of duplicating it twice? (I'm just guessing that'll work, but it may not). Same for below if true

Reply:
let me try this one, really good suggestion though, I used this format since this is the format when we were working on vault-k8s-fips image, but lets give it a try!

Reply:
it worked!